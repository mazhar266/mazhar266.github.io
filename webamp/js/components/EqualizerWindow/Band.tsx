import React from "react";
import { connect } from "react-redux";
import Slider from "rc-slider";
import { SET_BAND_FOCUS, UNSET_FOCUS } from "../../actionTypes";
import { AppState, Dispatch, Slider as SliderType } from "../../types";

interface StateProps {
  value: number;
  backgroundPosition: string;
}
interface DispatchProps {
  handleMouseDown(): void;
  handleMouseUp(): void;
}
interface OwnProps {
  id: string;
  band: SliderType;
  onChange(value: number): void;
}

type Props = StateProps & DispatchProps & OwnProps;

const MAX_VALUE = 100;
// Given a value between 1-100, return the sprite number (0-27)
export const spriteNumber = (value: number): number => {
  const percent = value / 100;
  return Math.round(percent * 27);
};

// Given a sprite number, return the x,y
export const spriteOffsets = (number: number): { x: number; y: number } => {
  const x = number % 14;
  const y = Math.floor(number / 14);
  return { x, y };
};

const Handle = () => <div className="rc-slider-handle" />;

const Band = ({
  value,
  backgroundPosition,
  id,
  onChange,
  handleMouseDown,
  handleMouseUp,
}: Props) => (
  <div id={id} className="band" style={{ backgroundPosition }}>
    <Slider
      min={0}
      max={MAX_VALUE}
      step={1}
      value={MAX_VALUE - value}
      vertical
      onChange={onChange}
      onBeforeChange={handleMouseDown}
      onAfterChange={handleMouseUp}
      handle={Handle}
    />
  </div>
);

const mapStateToProps = (state: AppState, ownProps: OwnProps): StateProps => {
  const value = state.equalizer.sliders[ownProps.band];
  const { x, y } = spriteOffsets(spriteNumber(value));
  const xOffset = x * 15; // Each sprite is 15px wide
  const yOffset = y * 65; // Each sprite is 15px tall
  const backgroundPosition = `-${xOffset}px -${yOffset}px`;
  return { value, backgroundPosition };
};

const mapDispatchToProps = (
  dispatch: Dispatch,
  ownProps: OwnProps
): DispatchProps => ({
  handleMouseDown: () =>
    dispatch({ type: SET_BAND_FOCUS, input: "eq", bandFocused: ownProps.band }),
  handleMouseUp: () => dispatch({ type: UNSET_FOCUS }),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Band);
