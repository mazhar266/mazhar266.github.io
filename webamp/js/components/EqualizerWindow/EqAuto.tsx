import React from "react";
import { connect } from "react-redux";
import classnames from "classnames";

import { SET_EQ_AUTO } from "../../actionTypes";
import { Dispatch, AppState } from "../../types";

interface StateProps {
  auto: boolean;
}

interface DispatchProps {
  toggleAuto(): void;
}

const EqAuto = (props: StateProps & DispatchProps) => {
  const className = classnames({ selected: props.auto });
  return <div id="auto" className={className} onClick={props.toggleAuto} />;
};

const mapStateToProps = (state: AppState): StateProps => {
  return { auto: state.equalizer.auto };
};
const mapDispatchToProps = () => (dispatch: Dispatch): DispatchProps => {
  // We don't support auto.
  return {
    toggleAuto: () => dispatch({ type: SET_EQ_AUTO, value: false }),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EqAuto);
