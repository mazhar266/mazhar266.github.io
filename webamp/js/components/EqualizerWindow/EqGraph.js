import React from "react";
import { connect } from "react-redux";
import { percentToRange, clamp } from "../../utils";
import { BANDS } from "../../constants";
import spline from "./spline";
import * as Selectros from "../../selectors";

const GRAPH_HEIGHT = 19;
const GRAPH_WIDTH = 113;

class EqGraph extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    this.canvasCtx = this.canvas.getContext("2d");
    this.width = Number(this.canvas.width);
    this.height = Number(this.canvas.height);

    this.createColorPattern(this.props.lineColorsImagePromise);
    this.createPreampLineImage(this.props.preampLineImagePromise);
  }

  componentDidUpdate() {
    this.canvasCtx.clearRect(0, 0, this.width, this.height);
    this.drawPreampLine();
    this.drawEqLine(); // This should paint on top of the preamp line
  }

  shouldComponentUpdate(nextProps) {
    if (
      this.props.lineColorsImagePromise !== nextProps.lineColorsImagePromise
    ) {
      this.createColorPattern(nextProps.lineColorsImagePromise);
    }
    if (
      this.props.preampLineImagePromise !== nextProps.preampLineImagePromise
    ) {
      this.createPreampLineImage(nextProps.preampLineImagePromise);
    }
    return true;
  }

  async createPreampLineImage(preampLineImagePromise) {
    const preampLineImg = await preampLineImagePromise;
    this.setState({ preampLineImg });
  }

  async createColorPattern(lineColorsImagePromise) {
    this.setState({
      colorPattern: this.canvasCtx.createPattern(
        await lineColorsImagePromise,
        "repeat-x"
      ),
    });
  }

  drawEqLine() {
    if (!this.state.colorPattern) {
      // The skin has not finished loading yet
      return;
    }
    const { props } = this;
    const amplitudes = BANDS.map(band => props[band]);

    this.canvasCtx.fillStyle = this.state.colorPattern;
    const paddingLeft = 2; // TODO: This should be 1.5

    const min = 0;
    const max = GRAPH_HEIGHT - 1;

    const xs = [];
    const ys = [];
    amplitudes.forEach((value, i) => {
      const percent = (100 - value) / 100;
      // Each band is 12 pixels widex
      xs.push(i * 12);
      ys.push(percentToRange(percent, min, max));
    });

    const allYs = spline(xs, ys);

    const maxX = xs[xs.length - 1];
    let lastY = ys[0];
    for (let x = 0; x <= maxX; x++) {
      const y = clamp(Math.round(allYs[x]), 0, GRAPH_HEIGHT - 1);
      const yTop = Math.min(y, lastY);
      const height = 1 + Math.abs(lastY - y);
      this.canvasCtx.fillRect(paddingLeft + x, yTop, 1, height);
      lastY = y;
    }
  }

  drawPreampLine() {
    const { preampLineImg } = this.state;
    if (!preampLineImg) {
      // The skin has not finished loading yet
      return;
    }
    const preampValue = percentToRange(
      this.props.preamp / 100,
      0,
      GRAPH_HEIGHT - 1
    );
    this.canvasCtx.drawImage(
      this.state.preampLineImg,
      0,
      preampValue,
      preampLineImg.width,
      preampLineImg.height
    );
  }

  render() {
    return (
      <canvas
        id="eqGraph"
        ref={node => (this.canvas = node)}
        width={GRAPH_WIDTH}
        height={GRAPH_HEIGHT}
      />
    );
  }
}

export default connect(state => ({
  ...state.equalizer.sliders,
  // WebampLazy.skinIsLoaded() makes an effort to ensure that these promises are
  // already resolved before our initial render. This means our setStates
  // should get called on a microtask and we should rerender with this loaded
  // before we paint.
  // This does not work when loading skins after initial render.
  lineColorsImagePromise: Selectros.getLineColorsImage(state),
  preampLineImagePromise: Selectros.getPreampLineImage(state),
}))(EqGraph);
