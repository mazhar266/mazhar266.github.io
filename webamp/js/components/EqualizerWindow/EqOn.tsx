import React from "react";
import { connect } from "react-redux";
import classnames from "classnames";

import { toggleEq } from "../../actionCreators";
import { AppState, Dispatch } from "../../types";

interface StateProps {
  on: boolean;
}

interface DispatchProps {
  toggleEq(): void;
}

const EqOn = (props: StateProps & DispatchProps) => {
  return (
    <div
      id="on"
      className={classnames({
        selected: props.on,
      })}
      onClick={props.toggleEq}
    />
  );
};

const mapStateToProps = (state: AppState): StateProps => ({
  on: state.equalizer.on,
});

const mapDispatchProps = (dispatch: Dispatch): DispatchProps => {
  return { toggleEq: () => dispatch(toggleEq()) };
};

export default connect(
  mapStateToProps,
  mapDispatchProps
)(EqOn);
