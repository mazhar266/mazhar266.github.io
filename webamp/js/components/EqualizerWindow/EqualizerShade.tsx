import React from "react";
import { connect } from "react-redux";
import Volume from "../Volume";
import Balance from "../Balance";
import { segment } from "../../utils";

import { closeWindow, toggleEqualizerShadeMode } from "../../actionCreators";
import { AppState, Dispatch } from "../../types";

interface StateProps {
  volume: number;
  balance: number;
}

interface DispatchProps {
  closeWindow(): void;
  toggleEqualizerShadeMode(): void;
}

const EqualizerShade = (props: StateProps & DispatchProps) => {
  const { volume, balance } = props;

  const classes = ["left", "center", "right"];
  const eqVolumeClassName = segment(0, 100, volume, classes);
  const eqBalanceClassName = segment(-100, 100, balance, classes);
  return (
    <div
      className="draggable"
      onDoubleClick={props.toggleEqualizerShadeMode}
      style={{ width: "100%", height: "100%" }}
    >
      <div id="equalizer-shade" onClick={props.toggleEqualizerShadeMode} />
      <div id="equalizer-close" onClick={props.closeWindow} />
      <Volume id="equalizer-volume" className={eqVolumeClassName} />
      <Balance id="equalizer-balance" className={eqBalanceClassName} />
    </div>
  );
};

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => {
  return {
    closeWindow: () => dispatch(closeWindow("equalizer")),
    toggleEqualizerShadeMode: () => dispatch(toggleEqualizerShadeMode()),
  };
};

const mapStateToProps = (state: AppState): StateProps => ({
  volume: state.media.volume,
  balance: state.media.balance,
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EqualizerShade);
