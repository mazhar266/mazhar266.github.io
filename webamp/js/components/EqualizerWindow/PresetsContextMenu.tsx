import React from "react";
import { connect } from "react-redux";
import builtin from "../../../presets/builtin.json";
import {
  openEqfFileDialog,
  downloadPreset,
  setEqFromObject,
} from "../../actionCreators";
import { Node, Parent, Hr } from "../ContextMenu";
import ContextMenuTarget from "../ContextMenuTarget";
import { Dispatch, EqfPreset } from "../../types";

interface DispatchProps {
  openEqfFileDialog(): void;
  downloadPreset(): void;
  setEqFromObject(preset: EqfPreset): void;
}

const PresetsContextMenu = (props: DispatchProps) => (
  <ContextMenuTarget top id="presets-context" handle={<div id="presets" />}>
    <Parent label="Load">
      {builtin.presets.map(preset => (
        <Node
          key={preset.name}
          onClick={() => props.setEqFromObject(preset)}
          label={preset.name}
        />
      ))}
      <Hr />
      <Node onClick={props.openEqfFileDialog} label="From Eqf..." />
    </Parent>
    <Node onClick={props.downloadPreset} label="Save" />
  </ContextMenuTarget>
);

const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => {
  return {
    openEqfFileDialog: () => dispatch(openEqfFileDialog()),
    downloadPreset: () => dispatch(downloadPreset()),
    setEqFromObject: preset => dispatch(setEqFromObject(preset)),
  };
};

export default connect(
  null,
  mapDispatchToProps
)(PresetsContextMenu);
