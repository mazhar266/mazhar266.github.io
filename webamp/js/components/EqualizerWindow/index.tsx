import React from "react";
import { connect } from "react-redux";
import classnames from "classnames";

import { BANDS, WINDOWS } from "../../constants";
import {
  setEqBand,
  setPreamp,
  setEqToMax,
  setEqToMid,
  setEqToMin,
  closeWindow,
  toggleEqualizerShadeMode,
} from "../../actionCreators";

import { getWindowShade } from "../../selectors";

import Band from "./Band";
import EqOn from "./EqOn";
import EqAuto from "./EqAuto";
import EqGraph from "./EqGraph";
import PresetsContextMenu from "./PresetsContextMenu";
import EqualizerShade from "./EqualizerShade";

import "../../../css/equalizer-window.css";
import { Dispatch, Band as BandType, AppState } from "../../types";
import FocusTarget from "../FocusTarget";

interface StateProps {
  doubled: boolean;
  selected: boolean;
  shade: boolean | undefined;
}

interface DispatchProps {
  setPreampValue(value: number): void;
  setEqToMin(): void;
  setEqToMid(): void;
  setEqToMax(): void;
  setHertzValue(hertz: BandType): (value: number) => void;
  closeEqualizerWindow(): void;
  toggleEqualizerShadeMode(): void;
}

const bandClassName = (band: BandType) => `band-${band}`;

const EqualizerWindow = (props: StateProps & DispatchProps) => {
  const { doubled, selected, shade } = props;

  const className = classnames({
    selected,
    doubled,
    shade,
    window: true,
    draggable: true,
  });
  return (
    <div id="equalizer-window" className={className}>
      <FocusTarget windowId={WINDOWS.EQUALIZER}>
        {props.shade ? (
          <React.StrictMode>
            <EqualizerShade />
          </React.StrictMode>
        ) : (
          <div>
            <React.StrictMode>
              <div
                className="equalizer-top title-bar draggable"
                onDoubleClick={props.toggleEqualizerShadeMode}
              >
                <div
                  id="equalizer-shade"
                  onClick={props.toggleEqualizerShadeMode}
                />
                <div
                  id="equalizer-close"
                  onClick={props.closeEqualizerWindow}
                />
              </div>
              <EqOn />
              <EqAuto />
              <EqGraph />
              <PresetsContextMenu />
            </React.StrictMode>
            <Band id="preamp" band="preamp" onChange={props.setPreampValue} />
            <div id="plus12db" onClick={props.setEqToMax} />
            <div id="zerodb" onClick={props.setEqToMid} />
            <div id="minus12db" onClick={props.setEqToMin} />
            {BANDS.map(hertz => (
              <Band
                key={hertz}
                id={bandClassName(hertz)}
                band={hertz}
                onChange={props.setHertzValue(hertz)}
              />
            ))}
          </div>
        )}
      </FocusTarget>
    </div>
  );
};

// This does not use the shorthand object syntax becuase `setHertzValue` needs
// to return a function.
const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
  setPreampValue: (value: number) => dispatch(setPreamp(value)),
  setEqToMin: () => dispatch(setEqToMin()),
  setEqToMid: () => dispatch(setEqToMid()),
  setEqToMax: () => dispatch(setEqToMax()),
  setHertzValue: (hertz: BandType) => (value: number) =>
    dispatch(setEqBand(hertz, value)),
  closeEqualizerWindow: () => dispatch(closeWindow("equalizer")),
  toggleEqualizerShadeMode: () => dispatch(toggleEqualizerShadeMode()),
});

const mapStateToProps = (state: AppState): StateProps => ({
  doubled: state.display.doubled,
  selected: state.windows.focused === WINDOWS.EQUALIZER,
  shade: getWindowShade(state)("equalizer"),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EqualizerWindow);
