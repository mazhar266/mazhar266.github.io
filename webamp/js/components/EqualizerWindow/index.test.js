import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import getStore from "../../store";

import EqualizerWindow from "./index";
const media = {
  addEventListener: jest.fn(),
  setVolume: jest.fn(),
  setPreamp: jest.fn(),
  setBalance: jest.fn(),
  getAnalyser: () => null,
  on: jest.fn(),
};

const canvasMockify = require("canvas-mock");

function createNodeMock(element) {
  if (element.type === "canvas") {
    const Canvas = window.document.createElement("canvas");
    canvasMockify(Canvas); // mock canvas functions required by Phaser.js are added
    return Canvas;
  }
  return null;
}

describe("EqualizerWindow", () => {
  let store;
  beforeEach(() => {
    store = getStore(media);
  });

  it("renders to snapshot", () => {
    const options = { createNodeMock };
    const tree = renderer
      .create(
        <Provider store={store}>
          <EqualizerWindow />
        </Provider>,
        options
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
