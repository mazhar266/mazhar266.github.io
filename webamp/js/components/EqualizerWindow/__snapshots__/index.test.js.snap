// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EqualizerWindow renders to snapshot 1`] = `
<div
  className="window draggable"
  id="equalizer-window"
>
  <div
    onFocus={[Function]}
    onMouseDown={[Function]}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
    tabIndex={-1}
  >
    <div>
      <div
        className="equalizer-top title-bar draggable"
        onDoubleClick={[Function]}
      >
        <div
          id="equalizer-shade"
          onClick={[Function]}
        />
        <div
          id="equalizer-close"
          onClick={[Function]}
        />
      </div>
      <div
        className="selected"
        id="on"
        onClick={[Function]}
      />
      <div
        className=""
        id="auto"
        onClick={[Function]}
      />
      <canvas
        height={19}
        id="eqGraph"
        width={113}
      />
      <div
        id="presets-context"
      >
        <div
          className="handle"
          onClick={[Function]}
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <div
            id="presets"
          />
        </div>
      </div>
      <div
        className="band"
        id="preamp"
        style={
          Object {
            "backgroundPosition": "-0px -65px",
          }
        }
      >
        <div
          className="rc-slider rc-slider-vertical"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="rc-slider-rail"
            style={Object {}}
          />
          <div
            className="rc-slider-track"
            style={
              Object {
                "bottom": "0%",
                "height": "50%",
              }
            }
          />
          <div
            className="rc-slider-step"
          />
          <div
            className="rc-slider-handle"
          />
          <div
            className="rc-slider-mark"
          />
        </div>
      </div>
      <div
        id="plus12db"
        onClick={[Function]}
      />
      <div
        id="zerodb"
        onClick={[Function]}
      />
      <div
        id="minus12db"
        onClick={[Function]}
      />
      <div
        className="band"
        id="band-60"
        style={
          Object {
            "backgroundPosition": "-0px -65px",
          }
        }
      >
        <div
          className="rc-slider rc-slider-vertical"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="rc-slider-rail"
            style={Object {}}
          />
          <div
            className="rc-slider-track"
            style={
              Object {
                "bottom": "0%",
                "height": "50%",
              }
            }
          />
          <div
            className="rc-slider-step"
          />
          <div
            className="rc-slider-handle"
          />
          <div
            className="rc-slider-mark"
          />
        </div>
      </div>
      <div
        className="band"
        id="band-170"
        style={
          Object {
            "backgroundPosition": "-0px -65px",
          }
        }
      >
        <div
          className="rc-slider rc-slider-vertical"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="rc-slider-rail"
            style={Object {}}
          />
          <div
            className="rc-slider-track"
            style={
              Object {
                "bottom": "0%",
                "height": "50%",
              }
            }
          />
          <div
            className="rc-slider-step"
          />
          <div
            className="rc-slider-handle"
          />
          <div
            className="rc-slider-mark"
          />
        </div>
      </div>
      <div
        className="band"
        id="band-310"
        style={
          Object {
            "backgroundPosition": "-0px -65px",
          }
        }
      >
        <div
          className="rc-slider rc-slider-vertical"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="rc-slider-rail"
            style={Object {}}
          />
          <div
            className="rc-slider-track"
            style={
              Object {
                "bottom": "0%",
                "height": "50%",
              }
            }
          />
          <div
            className="rc-slider-step"
          />
          <div
            className="rc-slider-handle"
          />
          <div
            className="rc-slider-mark"
          />
        </div>
      </div>
      <div
        className="band"
        id="band-600"
        style={
          Object {
            "backgroundPosition": "-0px -65px",
          }
        }
      >
        <div
          className="rc-slider rc-slider-vertical"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="rc-slider-rail"
            style={Object {}}
          />
          <div
            className="rc-slider-track"
            style={
              Object {
                "bottom": "0%",
                "height": "50%",
              }
            }
          />
          <div
            className="rc-slider-step"
          />
          <div
            className="rc-slider-handle"
          />
          <div
            className="rc-slider-mark"
          />
        </div>
      </div>
      <div
        className="band"
        id="band-1000"
        style={
          Object {
            "backgroundPosition": "-0px -65px",
          }
        }
      >
        <div
          className="rc-slider rc-slider-vertical"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="rc-slider-rail"
            style={Object {}}
          />
          <div
            className="rc-slider-track"
            style={
              Object {
                "bottom": "0%",
                "height": "50%",
              }
            }
          />
          <div
            className="rc-slider-step"
          />
          <div
            className="rc-slider-handle"
          />
          <div
            className="rc-slider-mark"
          />
        </div>
      </div>
      <div
        className="band"
        id="band-3000"
        style={
          Object {
            "backgroundPosition": "-0px -65px",
          }
        }
      >
        <div
          className="rc-slider rc-slider-vertical"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="rc-slider-rail"
            style={Object {}}
          />
          <div
            className="rc-slider-track"
            style={
              Object {
                "bottom": "0%",
                "height": "50%",
              }
            }
          />
          <div
            className="rc-slider-step"
          />
          <div
            className="rc-slider-handle"
          />
          <div
            className="rc-slider-mark"
          />
        </div>
      </div>
      <div
        className="band"
        id="band-6000"
        style={
          Object {
            "backgroundPosition": "-0px -65px",
          }
        }
      >
        <div
          className="rc-slider rc-slider-vertical"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="rc-slider-rail"
            style={Object {}}
          />
          <div
            className="rc-slider-track"
            style={
              Object {
                "bottom": "0%",
                "height": "50%",
              }
            }
          />
          <div
            className="rc-slider-step"
          />
          <div
            className="rc-slider-handle"
          />
          <div
            className="rc-slider-mark"
          />
        </div>
      </div>
      <div
        className="band"
        id="band-12000"
        style={
          Object {
            "backgroundPosition": "-0px -65px",
          }
        }
      >
        <div
          className="rc-slider rc-slider-vertical"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="rc-slider-rail"
            style={Object {}}
          />
          <div
            className="rc-slider-track"
            style={
              Object {
                "bottom": "0%",
                "height": "50%",
              }
            }
          />
          <div
            className="rc-slider-step"
          />
          <div
            className="rc-slider-handle"
          />
          <div
            className="rc-slider-mark"
          />
        </div>
      </div>
      <div
        className="band"
        id="band-14000"
        style={
          Object {
            "backgroundPosition": "-0px -65px",
          }
        }
      >
        <div
          className="rc-slider rc-slider-vertical"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="rc-slider-rail"
            style={Object {}}
          />
          <div
            className="rc-slider-track"
            style={
              Object {
                "bottom": "0%",
                "height": "50%",
              }
            }
          />
          <div
            className="rc-slider-step"
          />
          <div
            className="rc-slider-handle"
          />
          <div
            className="rc-slider-mark"
          />
        </div>
      </div>
      <div
        className="band"
        id="band-16000"
        style={
          Object {
            "backgroundPosition": "-0px -65px",
          }
        }
      >
        <div
          className="rc-slider rc-slider-vertical"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="rc-slider-rail"
            style={Object {}}
          />
          <div
            className="rc-slider-track"
            style={
              Object {
                "bottom": "0%",
                "height": "50%",
              }
            }
          />
          <div
            className="rc-slider-step"
          />
          <div
            className="rc-slider-handle"
          />
          <div
            className="rc-slider-mark"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
